features/netsocket/nsapi_dns.h

```diff
@@ -228,6 +228,14 @@ nsapi_error_t nsapi_dns_query_async_cancel(nsapi_error_t id);
  */
 void nsapi_dns_call_in_set(call_in_callback_cb_t callback);
 
+/**
+ * @brief nsapi_dns_reset Resets all internal states and frees reserved memory, see NOTE!
+ * Can be used to clean up system resources when there is no need for network connections.
+ * NOTE: Does NOT clear asynchronous ongoing operations!
+ * Currently only cleans up DNS cache (if used)
+ */
+void nsapi_dns_reset();
```

Comment:
Why is this in the public API?

I don't see any real use cases for resetting the cache, other than to test it.

Reply:
We don't have, but some application developer might want to use it so why should we prevent that?

Reply:
The real solution should be that when last NetworkStack is freed, we should free all DNS related stuff as well. Or if permanent DNS cache is needed, we could flag that.

Reply:
Such automatic purging should be optional. Cached information is still valid even if a network interface goes down.

No objection to adding a public API for test and/or application use though.

Reply:
Our DNS greentea tests use cache, even though perhaps sometimes they shouldn't. For example, if you run the whole suite, then the `_SIMULTANOUS` tests which are supposed to run 5 DNS requests at the same time, will in fact take three addresses resolved in previous tests out of cache and only open two more sockets to resolve the non-cached addresses. 
With this API in place we can clear cache where we don't want the test to rely on it.